pdc/apps/auth/views.py

```diff
@@ -903,3 +911,204 @@ def destroy(self, request, *args, **kwargs):
         On success, HTTP status code is 204 and the response has no content.
         """
         return super(GroupResourcePermissionViewSet, self).destroy(request, *args, **kwargs)
+
+
+class APIResourcePermissionViewSet(StrictQueryParamMixin,
+                                   ChangeSetCreateModelMixin,
+                                   ChangeSetDestroyModelMixin,
+                                   mixins.RetrieveModelMixin,
+                                   mixins.ListModelMixin,
+                                   MultiLookupFieldMixin,
+                                   viewsets.GenericViewSet):
+    """
+    This end-point provides API resource permissions.
+    """
+    queryset = models.GroupResourcePermission.objects.all()
+    serializer_class = serializers.GroupResourcePermissionSerializer
+
+    permission_classes = (APIPermission,)
+    lookup_fields = (
+        ('resource__name', r'[^/]+'),
+        ('permission__name', r'[^/]+'),
+    )
+    filter_class = filters.GroupResourcePermissionFilter
+
+    def _get_users_and_groups(self, resource_permission):
+        members = dict()
+        try:
+            group_resource_permission_list = get_list_or_404(models.GroupResourcePermission,
+                                                             resource_permission=resource_permission)
+            groups_list = [str(obj.group.name) for obj in group_resource_permission_list]
+        except Http404:
+            pass
+        # get all users
+        superusers_set = {user.username for user in models.User.objects.filter(is_superuser=True)}
+        users_set = {user.username for user in get_user_model().objects.filter(groups__name__in=groups_list)}
+        users_list = list(superusers_set.union(users_set))
+        members['groups'] = sorted(groups_list)
+        members['users'] = sorted(users_list)
+        return members
+
+    def list(self, request, *args, **kwargs):
+        """
+        Get information about API resource permissions.
+
+        __Method__: `GET`
+
+        __URL__: $LINK:apiresourcepermissions-list$
+
+        __Query params__:
+
+        * `resource` (string)
+        * `permission` (string)
+
+        __Response__:
+
+             # paged lists
+            {
+                "resource": string,
+                    {
+                        "permission": string,
+                            {
+                                "users": list,
+                                "groups" list
+                            }
+                             ...
+                    }
+                    ...
+            }
+
+        """
+        result = {}
```

Comment:
I don't quite understand why implement it here rather than in serializer. I am not quite experienced about code structure. But if you implement it here need to implement a lot of work serializer will do automatically, and I am not sure it support pagination, ordering, etc well.